core/controller/src/main/scala/whisk/core/controller/Actions.scala

```diff
@@ -604,6 +671,77 @@ trait WhiskActionsApi extends WhiskCollectionAPI {
         }
     }
 
+    /**
+     * helper method that checks the number of 'inlined' actions for a sequence is less than a threshold and checks recursion
+     * @param sequence the sequence to check
+     * @param components the components of the sequence
+     */
+    private def checkSequenceActionLimits(sequence: FullyQualifiedEntityName, components: Vector[FullyQualifiedEntityName]) (
+        implicit transid: TransactionId): Future[Boolean] = {
+        // traverse all actions and "inline" all actions that are sequence
+        // keep track of all sequences to detect recursion
+        info(this, s"Checking limits and recursion for actions in a sequence $sequence $components")
+        // first check that out of the box no more components than necessary
+        if (components.size > whiskConfig.actionSequenceLimit.toInt)
+            Future.failed(new TooManyActionsInSequence())
```

Comment:
I don't think it will? There is no else block, nothing will happen with the failed Future i believe.